### PR TITLE
(DOCSP-19552): Automatic client reset with seamless loss

### PR DESCRIPTION
**Pull request by mongodben**
## Pull Request Info

document new client reset with seamless loss method. update docs to include reference of how to do the previous method and also client reset with data recovery. 

### Jira

- https://jira.mongodb.org/browse/DOCSP-19552

### Staged Changes (Requires MongoDB Corp SSO)

*note: Node and React Native pages have all the same content*

- [Client Reset (Node)](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-19552/sdk/node/examples/reset-a-client-realm)
- [Client Reset (React Native)](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-19552/sdk/react-native/examples/reset-a-client-realm)
- [Manual Client Reset (Node)](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-19552/sdk/node/advanced/manual-client-reset)
- [Manual Client Reset (React Native)](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-19552/sdk/react-native/advanced/manual-client-reset)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/examples/generated/node/client-reset.codeblock.discard-unsynced-changes-after-destructive-schema-changes.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+// Once you have opened your Realm, you will have to keep a reference to it.
+// In the error handler, this reference is called `realm`
+async function handleSyncError(session, syncError) {
+  if (syncError.name == "ClientReset") {
```
Comment (kneth) [inline source/examples/generated/node/client-reset.codeblock.discard-unsynced-changes-after-destructive-schema-changes.js:4]:
```suggestion
  if (syncError.name === "ClientReset") {
```

**Review by kneth — APPROVED** [review]

**Review by cbush — APPROVED** [review]

File: examples/node/Examples/client-reset.js [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+      };
+      // :snippet-end:
+      Realm.open(config).then((openedRealm) => {
+        realm = openedRealm;
+        realm.write(() => {
+          realm.create("Doggo3", {
+            _id: new ObjectId(),
+            name: "Chippy",
+            age: 12,
+            _partition: "MyPartitionValue",
```
[the remark this answers is not in the document]
Comment (mongodben) [reply examples/node/Examples/client-reset.js:65]: TIL!

File: source/sdk/node/advanced/client-reset-data-recovery.txt [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 3
+   :class: singlecol
+
+This page explains how to manually recover unsynced realm data after a client reset 
+caused by a destructive schema change.
```
[the remark this answers is not in the document]
Comment (mongodben) [reply source/sdk/node/advanced/client-reset-data-recovery.txt:16]: oh good point forgot about that. (i think this PR was started before we made that decision). will update accordingly throughout PR. 

and no, it's not only for breaking schema changes here. good catch. removing here., but updating throughout

File: examples/node/Examples/client-reset.js [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+            // Download Realm from the server.
+            // Ensure that the backend state is fully downloaded before proceeding,
+            // which is the default behavior.
+            realm = await Realm.open(config);
+            // :remove-start:
+            expect(realm.isClosed).toBe(false);
+            resolve(true);
+            // :remove-end:
+            // :uncomment-start:
+            // realm.close();
```
[the remark this answers is not in the document]
Comment (mongodben) [reply examples/node/Examples/client-reset.js:131]: unsure. removing uncomment.

**Pull request merged**